File: 2015/day01/ruby/day01p2.rb
#2025 Levi D. Smith
f = open("../input01.txt")
while( strLine = f.gets )
  i = 0
  iValue = 0
  strLine.chars.each { 
    | c |
    case c[0]
    when '('
      iValue += 1 
    when ')'
      iValue -= 1
    end

    if (iValue == -1)
      puts(i + 1)
      break
    end
    i += 1
  }
end
f.close

File: 2015/day01/python/day01p2.py
#2025 Levi D. Smith
def main():
  with open("../input01.txt", "r") as f:
    strLine = f.readline()
    while (strLine != ''):
#      print(strLine)
      iValue = 0
      i = 0
      for char in strLine:
        if char == '(':
          iValue += 1
        elif char == ')':
          iValue -= 1

        if (iValue == -1):
          print(i + 1)
          break
        i += 1
      strLine = f.readline()


if __name__ == "__main__":
  main()
